plugins/inputs/fritzbox/fritzbox_test.go

```diff
@@ -0,0 +1,167 @@
+package fritzbox
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-tr064/mock"
+)
+
+func TestInitDefaults(t *testing.T) {
+	plugin := defaultFritzbox()
+	err := plugin.Init()
+	require.NoError(t, err)
+	require.Len(t, plugin.URLs, 0)
+	require.Equal(t, []string{"device", "wan", "ppp", "dsl", "wlan"}, plugin.Collect)
+	require.Equal(t, defaultTimeout, plugin.Timeout)
+	require.Empty(t, plugin.TLSCA)
+	require.Empty(t, plugin.TLSCert)
+	require.Empty(t, plugin.TLSKey)
+	require.Empty(t, plugin.TLSKeyPwd)
+	require.False(t, plugin.InsecureSkipVerify)
+}
+
+func TestConfig(t *testing.T) {
+	conf := config.NewConfig()
+	err := conf.LoadConfig("testdata/conf/fritzbox.conf")
+	require.NoError(t, err)
+	require.Len(t, conf.Inputs, 1)
+	plugin, ok := conf.Inputs[0].Input.(*Fritzbox)
+	require.True(t, ok)
+	require.Len(t, plugin.URLs, 2)
+	require.Equal(t, []string{"device", "wan", "ppp", "dsl", "wlan", "hosts"}, plugin.Collect)
+	require.Equal(t, config.Duration(60*time.Second), plugin.Timeout)
+	require.Equal(t, "secret", plugin.TLSKeyPwd)
+	require.True(t, plugin.InsecureSkipVerify)
+}
+
+const mockDocsDir = "testdata/mock"
```

Comment:
Please directly use the value in the respective spots.

Reply:
Adapted via 1ada7a8c36211e8f74afa86ed9638c45083686bb